### ISSUE TITLE
SQLite segv with virtual tables

### ISSUE THREAD

**Issue by rogerbinns** [rogerbinns/apsw#184]
Reported to me by Taylor C.

```
The attached Python script reliably causes a SIGSEGV for me on Ubuntu
14.04 (Python 2.7.6, sqlite3 3.8.2, apsw 3.8.2-r1), with the attached
stack trace.  The line in sqlite3.c is derived from vdbe.c for the
OP_VNext opcode:

   rc = pModule->xNext(pCur->pVtabCursor);

The cause of the SIGSEGV is that pModule is null.
```

```
import apsw
import sys

class Module(object):
    def Create(self, db, modulename, databasename, tablename, *args):
        print 'Module.Create'
        return 'CREATE TABLE t(x)', Table()
    def Destroy(self):
        print 'Module.Destroy'
        raise NotImplementedError
    Connect = Create
    Disconnect = Destroy

class Table(object):
    def BestIndex(self, _constraints, _orderbys):
        print 'Table.BestIndex'
        return None
    def Open(self):
        print 'Table.Open'
        return Cursor()
    def Destroy(self):
        print 'Table.Destroy'
        pass
    def UpdateInsertRow(self, _rowid, _fields):
        print 'Table.UpdateInsertRow'
        return 0
    def UpdateChangeRow(self, _rowid, _newrowid, _fields):
        print 'Table.UpdateChangeRow'
        pass
    def UpdateDeleteRow(self, _rowid):
        print 'Table.UpdateDeleteRow'
        pass

class Cursor(object):
    def Filter(self, _indexnum, _indexname, _constraintargs):
        print 'Cursor.Filter'
        self.idx = 0
    def Close(self):
        del self.idx
    def Column(self, i):
        print 'Cursor.Column'
        return self.idx
    def Eof(self):
        print 'Cursor.Eof'
        return self.idx == 1
    def Next(self):
        print 'Cursor.Next'
        self.idx += 1
    def Rowid(self):
        print 'Cursor.Rowid'
        return self.idx

db = apsw.Connection(':memory:')
db.createmodule('vtab', Module())
db.cursor().execute('CREATE VIRTUAL TABLE TEMP.vt USING vtab(x)')
db.cursor().execute('INSERT INTO TEMP.vt VALUES (42)')
cursor = db.cursor()
print 'select'
cursor.execute('SELECT * FROM TEMP.vt')
print 'drop'
db.cursor().execute('DROP TABLE TEMP.vt')
print 'iterate'
for x in cursor: print x
print 'done'
```

**Comment by rogerbinns** [comment]
This is the output of running it:

```
Module.Create
Table.UpdateInsertRow
select
Table.BestIndex
Table.Open
Cursor.Filter
Cursor.Eof
Cursor.Column
drop
Table.Destroy
iterate
(0,)
Segmentation fault
```

**Comment by rogerbinns** [comment]
There isn't anything I can realistically do about this, and it needs to be addressed within SQLite itself.  Details posted to SQLite at http://thread.gmane.org/gmane.comp.db.sqlite.general/93803

**Comment by rogerbinns** [comment]
The SQLite team fixed this today, and it will be in SQLite 3.8.9
